src/Test/Utilities/Shared/Compilation/CompilationExtensions.cs

```diff
@@ -148,5 +150,102 @@ internal static void VerifyAssemblyAliases(this Compilation compilation, params
 
             AssertEx.Equal(expectedAssembliesAndAliases, actual, itemInspector: s => '"' + s + '"');
         }
+
+        internal static void VerifyOperationTree(this Compilation compilation, string expectedOperationTree, bool skipImplicitlyDeclaredSymbols = false)
+        {
+            VerifyOperationTree(compilation, expectedOperationTree, symbolToVerify: null, skipImplicitlyDeclaredSymbols: skipImplicitlyDeclaredSymbols);
+        }
+
+        internal static void VerifyOperationTree(this Compilation compilation, string expectedOperationTree, string symbolToVerify, bool skipImplicitlyDeclaredSymbols = false)
+        {
+            var tree = compilation.SyntaxTrees.First();
+            var root = tree.GetRoot();
+            var model = compilation.GetSemanticModel(tree);
+            var declarations = new List<DeclarationInfo>();
+            model.ComputeDeclarationsInNode(root, getSymbol: true, builder: declarations, cancellationToken: CancellationToken.None);
+
+            var actualTextBuilder = new StringBuilder();
+            foreach (var declaration in declarations.Where(d => d.DeclaredSymbol != null).OrderBy(d => d.DeclaredSymbol.ToDisplayString()))
+            {
+                if (declaration.DeclaredNode == null ||
+                    declaration.DeclaredSymbol == null || 
```

Comment:
The first two checks seems redundant: `DeclaredNode` will never be null (there's an assert in the constructor) and elements with `DeclaredSymbol == null` are already filtered out from the IEnumerable